packages/shorebird_cli/lib/src/executables/shorebird_tools.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:io';
+
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/shorebird_process.dart';
+
+/// A reference to a [ShorebirdTools] instance.
+final shorebirdToolsRef = create(ShorebirdTools.new);
+
+/// The [ShorebirdTools] instance available in the current zone.
+ShorebirdTools get shorebirdTools => read(shorebirdToolsRef);
+
+/// {@template package_failed_exception}
+/// An exception thrown when a package fails.
+/// {@endtemplate}
+class PackageFailedException implements Exception {
+  /// {@macro package_failed_exception}
+  PackageFailedException(this.message);
+
+  final String message;
+
+  @override
+  String toString() => message;
+}
+
+/// A wrapper around the `shorebird_tools` executable.
+///
+/// Used to access many commands related to Shorebird's flutter tooling.
+class ShorebirdTools {
+  /// Returns if the the current flutter version supports this tool.
+  ///
+  /// This should be used to check if the tool is supported before running
+  /// any commands.
+  bool isSupported() {
+    return shorebirdToolsDirectory.existsSync();
+  }
+
+  Directory get shorebirdToolsDirectory {
+    final dir = Directory(
+      p.join(
+        shorebirdEnv.flutterDirectory.path,
+        'packages',
+        'shorebird_tools',
+      ),
+    );
+    logger.info(dir.path);
+    return dir;
+  }
+
+  Future<ShorebirdProcessResult> _run(List<String> args) {
+    return process.run(
+      shorebirdEnv.dartBinaryFile.path,
+      [
+        'run',
+        'shorebird_tools',
+        'package',
+        ...args,
+      ],
+      workingDirectory: shorebirdToolsDirectory.path,
+    );
+  }
+
+  Future<void> package({
+    required String patchPath,
+    required String outputPath,
+  }) async {
```

Comment:
A doc comment explaining this would be useful

Reply:
Should we enable the lint rule to enforce the docs of all public members? I guess it would make our life easier on reviews?

Reply:
I think we had that on at one point. Not sure why we turned it off, but I'd be in favor.

Reply:
cool. I can do that in a follow up

Reply:
I think we'll probably want to add a lot more doc comments before turning it on so we aren't totally overwhelmed with warnings, but sg!